### PR TITLE
feat(importer): Create Suite Importer page

### PR DESCRIPTION
**Pull request by atomantic**
## Summary

New on-ramp to the Create pipeline at `/importer`. Paste a finished short story / novel / screenplay / comic script + Universe Name + Series Name + content-type radio; the server runs three `heavy`-tier LLM passes (canon-extract + arc-extract in parallel, then issue-proposal seeded with the arc summary), returns a preview, and the Review phase lets the user toggle per-entry canon checkboxes + edit the arc + issue titles / synopses / arc roles / per-issue season pickers before committing.

On **commit**:
- `mergeExtractedBible` LWW-merges characters / places / objects into the universe (case-insensitive name match — preserves locked entries, never duplicates).
- `series.arc` + `series.seasons[]` are written.
- One issue is created per proposal with `stages.prose.output` seeded with the verbatim source excerpt and `stages.idea.input` seeded with the LLM's logline + synopsis (bundled into the initial `createIssue` payload — single disk write per issue).

**Find-or-create** on both universe + series names — re-running with the same names extends the existing world. Series-name match is scoped to its universe so a typo can't cross-link.

**Series-arc lock** is honored end-to-end (`analyzeImport` fails fast with `ERR_LOCKED` so heavy-tier tokens aren't burned on a doomed run; `commitImport` double-checks under the read window).

**v1 limits:** 200K-char hard cap on source size — chunked-extraction follow-up captured in PLAN.md.

**Wire-shape vs. storage-shape:** the prompts + Zod schemas + API use the forward-looking `places` label (matching the in-flight Phase 0b SETTING→PLACE rename); mapping to the on-disk `settings` field happens at the orchestrator boundary so the UI + LLM already speak the new vocabulary.

## Closes

This is the "Create Suite — Importer page" item from PLAN.md (queued by commit `357c0258`).

## Notes from the local code review

Two P0 bugs found in the local review and fixed before opening the PR:
1. **`returnsJson: true` missing** on `runStagedLLM` calls — `stageRunner` consults the per-call option (not the stage-config field), so the production LLM reply would have been a raw fenced string and every `Array.isArray(content?.field)` guard would silently fail. Tests passed because mocks pre-resolved to JS objects. Added regression-pin test.
2. **Liquid `{% if %}` syntax in the prompt templates** — PortOS's `applyTemplate` is Mustache-only, so the blocks passed through verbatim, leaking all four content-type variants into every prompt. Rewrote as `{{#isShortStory}}…{{/isShortStory}}` section guards and pass per-type booleans from the orchestrator. Added regression-pin test.

## Test plan

- [x] `npm test -- importer` from `server/` → 13 tests pass
- [x] Full server suite (`npm test` from `server/`) → 5011 tests pass, 5 skipped (LM Studio)
- [x] `npx vite build` from `client/` → builds clean; `Importer-*.js` chunk emits
- [ ] End-to-end manual: navigate `/importer`, paste a short story, Analyze → Review → Commit, confirm series + issues land in `/pipeline/series/:id`
- [ ] `⌘K → Importer` reaches the page
- [ ] Voice agent `ui_navigate "importer"` reaches the page (via `nav.create.importer`)
- [ ] Second-pass import with same universe+series names extends instead of duplicating
- [ ] Try with all four content types (short-story / novel / screenplay / comic-script) and verify the per-type guidance section renders correctly

## Deferred (captured in PLAN.md)

- Chunked extraction for source > 200K chars
- Auto-detect content type via small classifier prompt
- Replace-vs-append destructive-confirm for re-imports
- Route-level tests (`server/routes/importer.test.js`) for explicit JSON-contract coverage
- Swap inline `CanonCard` for shared component when "Phase 2 — Extract shared CanonCard" lands

### CODE REVIEW

File: server/services/importer.js [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+  const created = await createUniverse({ name });
+  return { universe: created, isExisting: false };
+}
+
+async function findOrCreateSeries(name, universeId) {
+  const found = await findSeriesByName(name, universeId);
+  if (found) return { series: found, isExisting: true };
+  const created = await createSeries({ name, universeId });
+  return { series: created, isExisting: false };
+}
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/services/importer.js]: Per PortOS CLAUDE.md, this is a single-user, single-instance app — no concurrent-request races. A double-click on Analyze is the only realistic vector, and even then the worst-case outcome is two universes/series with the same name that the user can see and clean up. Adding an in-flight Set guard would be cheap but isn't worth the complexity for an edge case the user can recognize and resolve manually. Leaving the simple find-then-create as-is.

File: server/services/importer.test.js [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+// namespace exotic object that `vi.importActual` returns is brittle: it
+// intercepts only `get`, can bypass `[[Module]]` invariants Vitest's transform
+// expects, and behaves unpredictably for `Symbol.toStringTag` / re-exports.
+vi.mock('../lib/fileUtils.js', async () => {
+  const actual = await vi.importActual('../lib/fileUtils.js');
+  return {
+    ...actual,
+    PATHS: { ...actual.PATHS, data: tempRoot },
+  };
+});
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/services/importer.test.js:45]: Acknowledged — this is theoretical given Vitest's per-test-file isolation and PortOS being single-user/single-instance per CLAUDE.md. The risk would only materialize if a sibling test file in the same worker imported a module that closed over `PATHS.data` at top level. Not aware of any such collision today; the fix (per-test mkdtemp + module-level reset) is non-trivial and out of scope for this PR. Leaving as-is — would revisit if a flaky-test pattern actually emerges.

File: client/src/services/apiImporter.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export const IMPORTER_CONTENT_TYPES = ['short-story', 'novel', 'screenplay', 'comic-script'];
+
+// Initial fallback for the source-char limit + arc-role list. These are the
+// values shipped with this client; on mount the Importer page fetches
+// `/importer/config` and overwrites them with whatever the server reports —
+// so a server-side bump (or a chunked-extraction follow-up that lifts the
+// cap) automatically reaches the UI without a client redeploy. Treat these
+// as last-resort defaults if the config call fails or is in flight.
+export const IMPORTER_SOURCE_CHAR_LIMIT_FALLBACK = 200_000;
+export const IMPORTER_ARC_ROLES_FALLBACK = ['pilot', 'complication', 'midpoint', 'b-plot', 'all-is-lost', 'finale'];
```
[the remark this answers is not in the document]
Comment (atomantic) [reply client/src/services/apiImporter.js:14]: Captured as a deferred [Centralise ARC_ROLES source-of-truth] item in PLAN.md — the same drift hazard applies to the prompt enum in importer-issue-proposal.md. Out of scope for v1 PR #253 (long-term maintainability, not a current bug).

File: server/services/importer.js [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+      canonSelections[selectionKey] || [],
+      kind,
+      { source: 'imported' },
+    ),
+  ]));
+  const updatedUniverse = await updateUniverse(universe.id, universePatch);
+
+  // sanitizeArc returns null on an entirely-empty payload — leave
+  // series.arc untouched in that case.
+  const sanitizedArc = sanitizeArc(arc);
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/services/importer.js:739]: Verified — `sanitizeArc(null)` is explicitly safe: see `server/lib/storyArc.js:238` (`if (raw == null) return null`). The `if (sanitizedArc || seasons.length > 0)` guard at the call site correctly short-circuits when sanitizeArc returns null. The existing `commitImport happy path with empty arc` test in `importer.test.js` exercises this branch end-to-end. Resolving — contract is already correct and tested.

File: server/lib/validation.js [hunk 5]
```diff
@@ -1055,3 +1056,108 @@ export const creativeDirectorSceneUpdateSchema = z.object({
...
+  seriesId: z.string().trim().min(1).max(120),
+  canonSelections: z.object({
+    characters: z.array(importerCanonEntry).max(BIBLE_LIMITS.ENTRIES_PER_BIBLE_MAX).default([]),
+    places: z.array(importerCanonEntry).max(BIBLE_LIMITS.ENTRIES_PER_BIBLE_MAX).default([]),
+    objects: z.array(importerCanonEntry).max(BIBLE_LIMITS.ENTRIES_PER_BIBLE_MAX).default([]),
+  }).default({ characters: [], places: [], objects: [] }),
+  arc: importerArcShape.nullable().optional(),
+  seasons: z.array(importerSeasonEntry).max(50).default([]),
+  issues: z.array(importerIssueEntry).min(1).max(50),
+}).strict();
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/lib/validation.js:1168]: Verified — the global Express JSON body limit is set to 55MB at `server/index.js:285` (`app.use(express.json({ limit: '55mb' }));`), which comfortably handles the ~25MB worst case (50 issues × 500K chars). No route-level override needed. Resolving.

File: client/src/pages/Importer.jsx [hunk 6]
```diff
@@ -0,0 +1,675 @@
...
+            value={a.shape || ''}
+            onChange={(e) => setArc({ ...a, shape: e.target.value || undefined })}
+            className="w-full bg-port-bg border border-port-border rounded px-3 py-2 text-sm focus:outline-none focus:border-port-accent"
+          >
+            <option value="">— pick one —</option>
+            {STORY_SHAPES.map((s) => (
+              <option key={s.id} value={s.id}>{s.label}</option>
+            ))}
+          </select>
+        </div>
```
[the remark this answers is not in the document]
Comment (atomantic) [reply client/src/pages/Importer.jsx:566]: Captured as a follow-up in PLAN.md — same drift hazard as the arc-roles centralisation item from round 3. Will be folded into the same `/importer/config` exposure work. Deferred from PR #253.

File: client/src/pages/Importer.jsx [hunk 7]
```diff
@@ -0,0 +1,675 @@
...
+      <div className="flex items-center justify-end gap-2 pt-2">
+        <button
+          type="submit"
+          disabled={!intakeValid || analyzing}
+          className="bg-port-accent hover:bg-port-accent/80 disabled:opacity-50 disabled:cursor-not-allowed text-white px-4 py-2 rounded text-sm font-medium flex items-center gap-2"
+        >
+          {analyzing ? <Loader2 className="w-4 h-4 animate-spin" /> : <FileInput className="w-4 h-4" />}
+          {analyzing ? 'Analyzing…' : 'Analyze'}
+        </button>
+      </div>
```
[the remark this answers is not in the document]
Comment (atomantic) [reply client/src/pages/Importer.jsx:381]: The Analyze button is already gated on `disabled={!intakeValid || analyzing}` (Importer.jsx:332) and the round-4 `<form onSubmit>` handler explicitly checks `if (intakeValid && !analyzing) onAnalyze()` before firing — so client-side double-click is guarded. Cross-tab races are explicitly out of scope per the round-1 acknowledgement (single-user app, single-server context — CLAUDE.md). Resolving.

File: data.sample/prompts/stages/importer-issue-proposal.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+## Per-content-type guidance for splitting
+
+{{#isShortStory}}
+A short story is typically **one issue**. Unless `targetIssueCount` is set to something larger, return exactly one issue covering the entire piece. The `proseExcerpt` is the whole source verbatim.
+{{/isShortStory}}
+{{#isNovel}}
+Novels usually map to **one issue per chapter** if chapters are short (≤3000 words), or chapter-pair groupings if chapters are longer. Honor the author's chapter breaks — they're the strongest structural signal. Aim for 6–15 issues for a typical novel.
+{{/isNovel}}
+{{#isScreenplay}}
+A standard screenplay is **one episode** (one issue). If the screenplay explicitly contains `ACT ONE` / `ACT TWO` / `ACT THREE` breaks AND the user has set `targetIssueCount > 1`, split on act breaks; otherwise return one issue.
```
[the remark this answers is not in the document]
Comment (atomantic) [reply data.sample/prompts/stages/importer-issue-proposal.md:36]: Captured in PLAN.md as a follow-up — fix needs both a new `isUserRequestedCount` flag flowing into the template and a reworded screenplay branch to gate the split correctly. Deferred from PR #253.

File: server/services/importer.js [hunk 9]
```diff
@@ -0,0 +1,628 @@
...
+    // stable, so only the issue creation re-runs.
+    const n = issues.length;
+    const partial = Object.assign(
+      new Error(
+        `The universe and series were updated successfully, but ${createdIssueIds.length} of ${n} issue${n === 1 ? '' : 's'} failed and were rolled back — retry to create the remaining issues. (Original error: ${issueErr.message})`,
+      ),
+      { code: ERR_PARTIAL_COMMIT_ISSUES },
+    );
+    throw partial;
+  }
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/services/importer.js:860]: Captured in PLAN.md as a follow-up — fix needs an API contract decision (tag the partial-commit response so retry can skip arc, OR version the arc field for optimistic concurrency). Deferred from PR #253.

File: scripts/migrations/015-importer-stage-prompts.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        await copyFile(samplePath, dataPath);
+        copied++;
+        console.log(`✅ seeded ${filename}`);
+      } catch (err) {
+        // Don't abort the batch — log and keep going so the other prompts
+        // still land. The operator sees exactly which one failed.
+        console.warn(`⚠️  importer-stage-prompts: copy failed for ${filename}: ${err.message}`);
+        skipped++;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (arthexis) [reply scripts/migrations/015-importer-stage-prompts.js:69]: Addressed in 4669c651a6fbe5b24c5ad20a63ad7fecc73322e9 by bypassing timestamp window filtering for offset-known unread bytes, with backlog regression coverage.

File: server/routes/importer.test.js [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+    const res = await request(app).post('/api/importer/commit').send({
+      universeId: 'missing',
+      seriesId: 'ser-1',
+      canonSelections: { characters: [], places: [], objects: [] },
+      arc: null,
+      seasons: [],
+      issues: [{ title: 'I1', arcPosition: 1 }],
+    });
+    expect(res.status).toBe(404);
+  });
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/routes/importer.test.js:166]: Confirmed: `server/services/universeBuilder.js` does export `ERR_NOT_FOUND` (it's the canonical not-found error code used by other consumers too). The route's `SERVICE_ERROR_STATUS[universeSvc.ERR_NOT_FOUND]` mapping is wired correctly. Round-5 also dropped the storyArc.js mock and round-6 dropped the universeBuilder/series mocks entirely so the real modules + real constants are now used end-to-end. Resolving.

File: server/services/importer.js [hunk 12]
```diff
@@ -0,0 +1,661 @@
...
+    '## Existing universe canon (do NOT duplicate these by name or aliases)',
+    '',
+    'This universe already has some canonical entries. Match by `name` (case-insensitive) **and by any listed `aliases`**. If an existing entry covers the same character / place / object, **omit it entirely** from your output — return only NEW entries. (Evidence for existing entries from this source is not currently re-merged; downstream evidence backfill is a follow-up.)',
+    '',
+    '```json',
+    json,
+    '```',
+    '',
+  ].join('\n');
+}
```
[the remark this answers is not in the document]
Comment (atomantic) [reply server/services/importer.js:228]: Confirmed safe — the template was refactored in round 4 to put `{{{existingCanonBlock}}}` at the top level OUTSIDE any `{{#section}}` wrapper (see `data.sample/prompts/stages/importer-canon-extract.md`). `promptTemplate.js` runs TRIPLE_RE substitution ONCE after the section-resolution loop has settled, so the substituted JSON value is never re-processed even if it contains literal `{{...}}` strings inside JSON-encoded user content. Resolving.
